### PR TITLE
Dgds add min max input

### PR DESCRIPTION
**Pull request by ckrogers**

### CODE REVIEW

File: hydroengine_service/dgds_functions.py [hunk 1]
```diff
@@ -26,6 +26,11 @@
 LAND = ee.Image('users/gena/land_polygons_image')
 LANDMASK = ee.Image(LAND.unmask(1, False).Not().resample('bicubic').focal_mode(2))
 
+def validate_min_lt_max(min, max):
+    min_lt_max = min < max
+    if not min_lt_max:
```
Comment (CindyvdVries) [inline hydroengine_service/dgds_functions.py:31]: if min > max

bit overdoing it with defining  min lt max and then validating the other way around?

File: hydroengine_service/dgds_functions.py [hunk 2]
```diff
@@ -162,6 +178,12 @@ def visualize_elevation(image,
     :param hillshade: Boolean for hillshading, default True
     :return: Google Earth Engine ee.Image() object, Hillshaded image
     """
+    # validate min is less than max, otherwise raise error
+    min = data_params['bathy_vis_params']['min']
+    max = data_params['topo_vis_params']['max']
+    if min and max is not None:
+        validate_min_lt_max(min, max)
```
Comment (CindyvdVries) [inline hydroengine_service/dgds_functions.py:185]: the function raises an error when min > max. 
Can you send that error message back as a response, that we know what went wrong?

File: hydroengine_service/dgds_functions.py [hunk 3]
```diff
@@ -250,11 +272,19 @@ def generate_elevation_map(dataset_list=None):
         dataset_list = ELEVATION_DATA.keys()
 
     mosaic_image = mosaic_elevation_datasets(dataset_list)
-
     data_params = DATASETS_VIS["projects/dgds-gee/bathymetry/gebco/2019"]
-    final_image = visualize_elevation(mosaic_image, data_params=data_params)
-    url = _get_gee_url(final_image)
 
+    if min:
```
Comment (CindyvdVries) [inline hydroengine_service/dgds_functions.py:277]: if min_range and if max_range?

File: hydroengine_service/dgds_functions.py [hunk 4]
```diff
@@ -59,6 +66,8 @@ def get_dgds_data(source,
     :param start_date: String, start date to filter collection on
     :param end_date: String, end date to filter collection on
     :param image_num_limit: String, limit of objects to return
+    :param min_range: Number,
```
Comment (CindyvdVries) [inline hydroengine_service/dgds_functions.py:69]: comments

File: hydroengine_service/dgds_functions.py [hunk 5]
```diff
@@ -80,7 +89,14 @@ def get_dgds_data(source,
         else:
             function = function.get(band, None)
 
-    image_info = _get_wms_url(returned_url_id, type=data_params['type'], band=band, function=function)
+    image_info = _get_wms_url(
+        image_id=returned_url_id,
+        type=data_params['type'],
+        band=band,
+        function=function,
+        min=min_range,
```
Comment (CindyvdVries) [inline hydroengine_service/dgds_functions.py:97]: confusing min and min_range use, can we call them all min ?

File: hydroengine_service/dgds_functions.py [hunk 6]
```diff
@@ -280,20 +310,25 @@ def generate_elevation_map(dataset_list=None):
...
-def visualize_gebco(source, band):
+def visualize_gebco(source, band, min=None, max=None):
     """
     Specialized function to visualize GEBCO data
     :param source: String, Google Earth Engine image id
     :param band: String, band of image to visualize
     :return: Dictionary
     """
     data_params = DATASETS_VIS[source]
+    if min:
```
Comment (CindyvdVries) [inline hydroengine_service/dgds_functions.py:321]: Aha! see last two comments, need to call them all the same!

File: hydroengine_service/dgds_functions.py (hunk 3 above)
Comment (CindyvdVries) [reply hydroengine_service/dgds_functions.py:277]: if min is not None
See comment on line 584 - # min/max values can be zero, should not be None

File: hydroengine_service/dgds_functions.py [hunk 7]
```diff
@@ -250,11 +272,19 @@ def generate_elevation_map(dataset_list=None):
...
 
     mosaic_image = mosaic_elevation_datasets(dataset_list)
-
     data_params = DATASETS_VIS["projects/dgds-gee/bathymetry/gebco/2019"]
-    final_image = visualize_elevation(mosaic_image, data_params=data_params)
-    url = _get_gee_url(final_image)
 
+    if min:
+        data_params['bathy_vis_params']['min'] = min_range
+    if max:
```
Comment (CindyvdVries) [inline hydroengine_service/dgds_functions.py:279]: if max is not None
See comment on line 584 - # min/max values can be zero, should not be None

File: hydroengine_service/dgds_functions.py (hunk 6 above)
Comment (CindyvdVries) [reply hydroengine_service/dgds_functions.py:321]: if min is not None
See comment on line 584 - # min/max values can be zero, should not be None

File: hydroengine_service/dgds_functions.py [hunk 8]
```diff
@@ -280,20 +310,25 @@ def generate_elevation_map(dataset_list=None):
...
     """
     Specialized function to visualize GEBCO data
     :param source: String, Google Earth Engine image id
     :param band: String, band of image to visualize
     :return: Dictionary
     """
     data_params = DATASETS_VIS[source]
+    if min:
+        data_params['bathy_vis_params']['min'] = min
+    if max:
```
Comment (CindyvdVries) [inline hydroengine_service/dgds_functions.py:323]: if max is not None
See comment on line 584 - # min/max values can be zero, should not be None

File: hydroengine_service/dgds_functions.py (hunk 1 above)
Comment (ckrogers) [reply hydroengine_service/dgds_functions.py]: Good point, will reduce

File: hydroengine_service/dgds_functions.py (hunk 2 above)
Comment (ckrogers) [reply hydroengine_service/dgds_functions.py:184]: the raise error_handler.InvalidUsage will do this for hydroengine (return 400 with message), we should check dgds backend passes this along

**Pull request merged**